### ISSUE TITLE
CarrotTab isAvailable fails with non-numeric buildId

### ISSUE THREAD

**Issue by ben-biddington** [ben-biddington/Coriander.Haarlem#8]
@see: http://youtrack.jetbrains.net/issue/TW-11465?query=in+use

**Comment by ben-biddington** [comment]
This happens when you use one of the "permalinks" at the bottom of any build screen. Other links include "Last successful build" and "Last finished build".

These links look like:

```
/viewLog.html?buildTypeId=bt5&buildId=lastPinned
```

rather than:

```
/viewLog.html?buildId=99&tab=buildResultsDiv&buildTypeId=bt5
```

Under these alias conditions there is a cached copy of the build at:

```
request.getAttribute("BUILD_KEY")
```

which needs to be used rather than:

```
buildServer.findBuildInstanceById(Long)
```

**Comment by ben-biddington** [comment]
This has been fixed in 3804454bb55262d43a954312e1ee7431d795bef7
